src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -294,6 +294,13 @@ private Request CreateNewRequest_NoLock()
 
         public override Task<T> GetValueAsync(CancellationToken cancellationToken)
```

Comment:
Been looking at `GetValue` in the same file.  It has the same pattern as `GetValueAsync` here.  Did we consider applying the same optimization of avoiding the lock? 

Avoiding the lock in general seems safe.  That or TryGetValue's been broken for some time.